runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/collections/CaseInsensitiveMap.kt

```diff
@@ -36,11 +36,11 @@ internal class CaseInsensitiveMap<Value> : MutableMap<String, Value> {
 
     override val entries: MutableSet<MutableMap.MutableEntry<String, Value>>
         get() = impl.entries.map {
-            Entry(it.key.s, it.value)
+            Entry(it.key.normalized, it.value)
         }.toMutableSet()
 
     override val keys: MutableSet<String>
-        get() = impl.keys.map { it.s }.toMutableSet()
+        get() = impl.keys.map { it.normalized }.toMutableSet()
```

Comment:
These changes mean that tests like the following will fail:
```
    @Test
    fun testSimpleKeyOp() {
        val map = CaseInsensitiveMap<String>()
        map["A"] = "apple"
        assertTrue(map.keys.contains("A"))
    }
```

Is that expected from a case insensitive map? I think the keys should remain unmodified, they should only be case-insensitive for equality operations.

Maybe both `map.keys.contains("A")` and `map.keys.contains("a")` should be true.

Reply:
Oh yes, they should both be true. Revision incoming...